Fix the link so it points to the correct docdir and gzipped manuall

--- a/doc/mutt.man
+++ b/doc/mutt.man
@@ -185,7 +185,7 @@
 System mapping between MIME types and file extensions.
 .IP "@bindir@/mutt_dotlock"
 The privileged dotlocking program.
-.IP "@docdir@/manual.txt"
+.IP "/usr/share/doc/mutt/manual.txt.gz"
 The Mutt manual.
 .SH BUGS
 .PP
